src/jit/importer.cpp

```diff
@@ -1951,6 +1951,12 @@ bool                Compiler::impSpillStackEntry(unsigned   level,
     /* Assign the spilled entry to the temp */
     impAssignTempGen(tnum, tree, verCurrentState.esStack[level].seTypeInfo.GetClassHandle(), level);
 
+    // If the tree contains var address on stack, propagate such flag to a local which is spilled.
+    if (tree->IsVarAddr())
+    {
+        lvaTable[tnum].lvIsVarAddrOnStack = true;
+    }
```

Comment:
Would this will address all possible ways in which the following IR form can occur

cast (double<-byref, op1 = lclvar of type byref)

For e.g. in IL itself could be producing such an IR pattern

ldloca.s 0  // load the address of a local
starg.0      // store to a byref arg
conv.r8     // convert byref arg to double